torch/_dynamo/testing.py

```diff
@@ -103,7 +103,7 @@ def reduce_to_scalar_loss(out):
     """Reduce the output of a model to get scalar loss"""
     if isinstance(out, torch.Tensor):
         # Mean does not work on integer tensors
-        return out.sum() / out.numel()
+        return out.sum() / torch.tensor(out.numel(), device=out.device)
```

Comment:
Why is this not captured the graph itself instead of lifted as an input ?

```
    def forward(self, arg0_1: "f16[]cuda:0", arg1_1: "i64[]cpu", 
```

Reply:
compiled autograd lifts all tensors as it traces the autograd nodes